rwatch/ui/animation/property_animation.c

```diff
@@ -0,0 +1,172 @@
+/* property_animation.c
+ * a convenience animation for 
+ * libRebbleOS
+ *
+ * Author: Carson Katri <me@carsonkatri.com>
+ */
+
+#include "librebble.h"
+#include "appmanager.h"
+#include "FreeRTOS.h"
+#include "property_animation.h"
+#include "animation.h"
+
+void property_animation_update_grect(PropertyAnimation * property_animation, const uint32_t distance_normalized)
+{
+    int progress = (distance_normalized * 100) / ANIMATION_NORMALIZED_MAX;
+    
+    Layer *layer = (Layer *) property_animation->subject;
+    
+    GRect *from = &property_animation->values.from.grect;
+    GRect *to = &property_animation->values.to.grect;
+    
+    GRect new_rect = GRect(from->origin.x + (((to->origin.x - from->origin.x) / 100.0)*progress), from->origin.y + (((to->origin.y - from->origin.y) / 100.0)*progress), from->size.w + (((to->size.w - from->size.w) / 100.0)*progress), from->size.h + (((to->size.h - from->size.h) / 100.0)*progress));
+    if (property_animation->impl.accessors.setter.grect)
+        property_animation->impl.accessors.setter.grect(layer, new_rect);
+    //layer_set_frame(layer, new_rect);
+    layer_mark_dirty(layer);
+}
+
+void property_animation_update_gpoint(PropertyAnimation * property_animation, const uint32_t distance_normalized)
+{
+    int progress = (distance_normalized * 100) / ANIMATION_NORMALIZED_MAX;
+    
+    Layer *layer = (Layer *) property_animation->subject;
+    
+    GRect *from = &property_animation->values.from.grect;
+    GRect *to = &property_animation->values.to.grect;
+    
+    GRect new_rect = *from;
+    new_rect.origin.x = from->origin.x + (((to->origin.x - from->origin.x) / 100.0)*progress);
```

Comment:
Similar in this case -- avoid floating point arith, by doing your interpolation as fixed point.